tools/shared/src/main/scala/org/scalajs/core/tools/javascript/ScalaJSClassEmitter.scala

```diff
@@ -66,6 +69,11 @@ final class ScalaJSClassEmitter private (
     linkingUnit.classDefs.map(c => c.encodedName -> c).toMap
   }
 
+  private[javascript] def isInterface(className: String): Boolean = {
+    linkingUnit != null &&
+    linkedClassByName(className).kind == ClassKind.Interface
```

Comment:
Yes, all sorts of stuff is completely shuffled around when a thing switches from class to interface and vice versa. In the incremental optimizer, classes and interfaces live in completely different namespaces, so they are necessarily recreated.

I can see, in theory, a failure, though. In theory, a third-party class would have an `ApplyStatically` to `C.foo`. If `C` switches from class to interface or the other way around, the IR does not prescribe that this call needs to change. Therefore, the result could be broken. However, in practice, only subclasses and subtraits can call statically (due to the rules of the Scala *language*), and in scalac, it makes an awful lot of difference internally whether you extend a class or mix in a trait/interface, which means it will always be recompiled, leading to a different input IR anyway.

There are 3 possible ways forward about this:
* Do nothing, and accept the theoretical flaw
* Throw away the emitter cache when an entity switches from class to interface or the other way around (should not happen often in practical situations)
* Use a different IR node for statically calling a default method, for example `ApplyDefaultStatically`